staker-rewards/test/rewards-calculator.test.js

```diff
@@ -164,4 +180,332 @@ describe("rewards calculator", async () => {
+        mockContext,
+        interval,
+        [operatorParameters]
+      )
+
+      const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+      assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), false)
+    }
+  )
+
+  it(
+    "should set total rewards to zero if keygenCount >= 10 and " +
+      "failures percentage exceeds 10%",
+    async () => {
+      const mockContext = createMockContext()
+
+      setupContractsMock(mockContext)
+
+      const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+      operatorParameters.operatorSLA.keygenCount = 10
+      operatorParameters.operatorSLA.keygenFailCount = 2
+      operatorParameters.operatorSLA.keygenSLA = 80
+
+      const rewardsCalculator = await RewardsCalculator.initialize(
+        mockContext,
+        interval,
+        [operatorParameters]
+      )
+
+      const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+      assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), true)
+    }
+  )
+
+  it(
+    "should NOT set total rewards to zero if keygenCount < 10 and " +
+      "there is no more than 1 failure",
+    async () => {
+      const mockContext = createMockContext()
+
+      setupContractsMock(mockContext)
+
+      const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+      operatorParameters.operatorSLA.keygenCount = 9
+      operatorParameters.operatorSLA.keygenFailCount = 1
+      operatorParameters.operatorSLA.keygenSLA = 88.88
+
+      const rewardsCalculator = await RewardsCalculator.initialize(
+        mockContext,
+        interval,
+        [operatorParameters]
+      )
+
+      const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+      assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), false)
+    }
+  )
+
+  it(
+    "should set total rewards to zero if keygenCount < 10 and " +
+      "there is more than 1 failure",
+    async () => {
+      const mockContext = createMockContext()
+
+      setupContractsMock(mockContext)
+
+      const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+      operatorParameters.operatorSLA.keygenCount = 9
+      operatorParameters.operatorSLA.keygenFailCount = 2
+      operatorParameters.operatorSLA.keygenSLA = 77.77
+
+      const rewardsCalculator = await RewardsCalculator.initialize(
+        mockContext,
+        interval,
+        [operatorParameters]
+      )
+
+      const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+      assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), true)
+    }
+  )
+
+  it(
+    "should NOT set total rewards to zero if signatureCount >= 20 and " +
+      "failures percentage doesn't exceed 5%",
+    async () => {
+      const mockContext = createMockContext()
+
+      setupContractsMock(mockContext)
+
+      const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+      operatorParameters.operatorSLA.signatureCount = 20
```

Comment:
Same question as for the key-gen case, can we have two separate tests: one for `== 20` and another for `>= 20`?